move loci parsing logic from Common into loci pkg

ryan-williams opened the pull request:
- I decided to put put the `LociArgs` → `LociParser` logic into `LociArgs.parse` instead of a `LociParser` constructor to keep `LociArgs` out of `LociParser.scala`, because the latter feels further down the stack than the former

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/hammerlab/guacamole/459)
<!-- Reviewable:end -->

e5c commented:
Reviewed 7 of 9 files at r1.
Review status: 7 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hammerlab/guacamole/459)*
<!-- Sent from Reviewable.io -->

e5c commented:
LGTM; I've made the small loci-IO fixes necessary for [issue-446](https://github.com/hammerlab/guacamole/issues/446) but can rebase once this has been merged 

---

Reviewed 3 of 9 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hammerlab/guacamole/459#-:-KGeAdfXU0Z9P2BxZ3s8:1431777812)*
<!-- Sent from Reviewable.io -->

ryan-williams commented:
Cool, thanks @e5c! I'm also happy to rebase anyones' changes as some of these get merged in since it may be disruptive; lmk if that would be helpful!

e5c commented:
![reviewable](https://cloud.githubusercontent.com/assets/6350854/14939916/039e3b4c-0f2b-11e6-941b-bb7b76d4f34d.png)
(Separately, not sure why it says above that I've only reviewed 3 of 9 files? Viewing Reviewable itself indicates otherwise AFAICT)

arahuja commented:
Review status: 3 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/main/scala/org/hammerlab/guacamole/commands/GermlineAssemblyCaller.scala, line 197 \[r2\]](https://reviewable.io:443/reviews/hammerlab/guacamole/459#-KGmCqbXPt_zhnAXW-9P:-KGmCqbXPt_zhnAXW-9Q:359707415) ([raw file](https://github.com/hammerlab/guacamole/blob/b53ce002678b1145b4fea48fde7c8f5027e8d56d/src/main/scala/org/hammerlab/guacamole/commands/GermlineAssemblyCaller.scala#L197)):*
Maybe just personal preference, but I would read this as parsing all of the args out of input rather than the loci. Maybe also just due to experience with Python

---

*[src/main/scala/org/hammerlab/guacamole/loci/LociArgs.scala, line 28 \[r2\]](https://reviewable.io:443/reviews/hammerlab/guacamole/459#-KGmD0X300T7NkvfCPmq:-KGmD0X300T7NkvfCPmr:2069081922) ([raw file](https://github.com/hammerlab/guacamole/blob/b53ce002678b1145b4fea48fde7c8f5027e8d56d/src/main/scala/org/hammerlab/guacamole/loci/LociArgs.scala#L28)):*
I know this was there before, but `default` is a fairly ambiguous​ argument name. But that might be due to how this is structured. This function operates on the class values or gives that default. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hammerlab/guacamole/459)*
<!-- Sent from Reviewable.io -->

ryan-williams commented:
Review status: 0 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/main/scala/org/hammerlab/guacamole/commands/GermlineAssemblyCaller.scala, line 197 \[r2\]](https://reviewable.io:443/reviews/hammerlab/guacamole/459#-KGmCqbXPt_zhnAXW-9P:-KGmXS7j7mYuj6wp1BA6:-959184998) ([raw file](https://github.com/hammerlab/guacamole/blob/b53ce002678b1145b4fea48fde7c8f5027e8d56d/src/main/scala/org/hammerlab/guacamole/commands/GermlineAssemblyCaller.scala#L197)):*
Good point; changed this to `args.parseLoci()`

---

*[src/main/scala/org/hammerlab/guacamole/loci/LociArgs.scala, line 28 \[r2\]](https://reviewable.io:443/reviews/hammerlab/guacamole/459#-KGmD0X300T7NkvfCPmq:-KGmYPHepm2UjTnV-jZz:1667718263) ([raw file](https://github.com/hammerlab/guacamole/blob/b53ce002678b1145b4fea48fde7c8f5027e8d56d/src/main/scala/org/hammerlab/guacamole/loci/LociArgs.scala#L28)):*
cool, changed this to `fallback` and added a method comment documenting the assumptions here

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hammerlab/guacamole/459)*
<!-- Sent from Reviewable.io -->

arahuja commented:
👍
